cmd/rpcdaemon/commands/get_receipts.go

```diff
@@ -139,3 +234,214 @@ func (api *APIImpl) GetTransactionReceipt(ctx context.Context, hash common.Hash)
+
+	if indexed := sections * size; indexed > uint64(f.begin) {
+		if indexed > end {
+			logs, err = f.indexedLogs(ctx, end, api)
+		} else {
+			logs, err = f.indexedLogs(ctx, indexed-1, api)
+		}
+		if err != nil {
+			return logs, err
+		}
+	}
+	rest, err := f.unindexedLogs(ctx, end, api)
+	logs = append(logs, rest...)
+	return logs, err
+}
+
+// indexedLogs returns the logs matching the filter criteria based on the bloom
+// bits indexed available locally or via the network.
+func (f *Filter) indexedLogs(ctx context.Context, end uint64, api *APIImpl) ([]*types.Log, error) {
+	// Iterate over the matches until exhausted or context closed
+	var logs []*types.Log
+
+	for num := f.begin; num < int64(end)+1; num++ {
+		// Retrieve the suggested block and pull any truly matching logs
+		header, err := api.GetHeaderByNumber(ctx, rpc.BlockNumber(num))
+		if header == nil || err != nil {
+			return logs, err
+		}
+		found, err := f.checkMatches(ctx, header, api)
+		if err != nil {
+			return logs, err
+		}
+		logs = append(logs, found...)
+	}
+
+	return logs, nil
+}
+
+// unindexedLogs returns the logs matching the filter criteria based on raw block
+// iteration and bloom matching.
+func (f *Filter) unindexedLogs(ctx context.Context, end uint64, api *APIImpl) ([]*types.Log, error) {
+	var logs []*types.Log
+
+	for ; f.begin <= int64(end); f.begin++ {
+		header, err := api.GetHeaderByNumber(ctx, rpc.BlockNumber(f.begin))
+		if header == nil || err != nil {
+			return logs, err
+		}
+		found, err := f.blockLogs(ctx, header, api)
+		if err != nil {
+			return logs, err
+		}
+		logs = append(logs, found...)
+	}
+	return logs, nil
+}
+
+// blockLogs returns the logs matching the filter criteria within a single block.
+func (f *Filter) blockLogs(ctx context.Context, header *types.Header, api *APIImpl) (logs []*types.Log, err error) {
+	if bloomFilter(header.Bloom, f.addresses, f.topics) {
+		found, err := f.checkMatches(ctx, header, api)
+		if err != nil {
+			return logs, err
+		}
+		logs = append(logs, found...)
+	}
+	return logs, nil
+}
+
+// checkMatches checks if the receipts belonging to the given header contain any log events that
+// match the filter criteria. This function is called when the bloom filter signals a potential match.
+func (f *Filter) checkMatches(ctx context.Context, header *types.Header, api *APIImpl) (logs []*types.Log, err error) {
+	// Get the logs of the block
+	logsList, err := api.GetLogsByHash(ctx, header.Hash())
+	if err != nil {
+		return nil, err
+	}
+	unfiltered := make([]*types.Log, 0, len(logsList))
+	for _, logs := range logsList {
+		unfiltered = append(unfiltered, logs...)
+	}
+	logs = filterLogs(unfiltered, nil, nil, f.addresses, f.topics)
+	if len(logs) > 0 {
+		// We have matching logs, check if we need to resolve full logs via the light client
+		if logs[0].TxHash == (common.Hash{}) {
+			receipts := rawdb.ReadReceipts(api.dbReader, header.Hash(), header.Number.Uint64(), params.MainnetChainConfig)
```

Comment:
Could you remove the hard-coded config? There is a way to read it from the DB (you can see in the `getReceipt` method or similar)

Reply:
Done